regolith/helpers/u_milestonehelper.py

```diff
@@ -110,161 +102,173 @@ def construct_global_ctx(self):
+                pra = fragment_retrieval(self.gtx["projecta"], ["_id"],
+                                         rc.projectum_id)
+                print("Projectum not found. Projecta with similar names: ")
+                for i in range(len(pra)):
+                    print(f"{pra[i].get('_id')}")
+                print("Please rerun the helper specifying the complete ID.\n"
+                      "If your prum id looks correct, check that this id is in the collection "
+                      "in the database that regolith is looking in (i.e., {rc.database}).\n"
+                      "If this is the case, rerun with --database set to the database where the item is located.")
+                return
+            milestones = deepcopy(target_prum.get('milestones'))
+            all_milestones = []
+            for item in milestones:
+                item['identifier'] = 'milestones'
+            all_milestones.extend(milestones)
+            for i in all_milestones:
+                i['due_date'] = get_due_date(i)
+            all_milestones.sort(key=lambda x: x['due_date'], reverse=False)
+            index_list = list(range(2, (len(all_milestones) + 2)))
+            mil = {}
+            if not rc.due_date or not rc.name or not rc.objective:
+                raise RuntimeError(
+                    "name, objective, and due date are required for a new milestone")
+            mil.update({'due_date': rc.due_date})
+            mil['due_date'] = get_due_date(mil)
+            mil.update({'objective': rc.objective, 'name': rc.name, 'uuid': get_uuid()})
+            if rc.audience:
+                mil.update({'audience': rc.audience})
+            else:
+                mil.update({'audience': ['lead', 'pi', 'group_members']})
+            if rc.status:
+                mil.update({'status': rc.status})
+            else:
+                mil.update({'status': 'proposed'})
+            if rc.notes:
+                mil.update({'notes': rc.notes})
+            if rc.type:
+                if rc.type in MILESTONE_TYPES:
                     mil.update({'type': rc.type})
                 else:
-                    mil.update({'type': 'meeting'})
-                if rc.status:
-                    mil.update({'status': rc.status})
-                else:
-                    mil.update({'status': 'proposed'})
-                if rc.notes:
-                    mil.update({'notes': rc.notes})
-                mil.update({'identifier': "milestones"})
-                new_mil.append(mil)
-                pdoc = {'milestones': mil}
-            if idx > 1:
-                doc = deepcopy(all_milestones[idx - 2])
-                identifier = doc['identifier']
-                if not doc.get(
-                        'type') and not rc.type and identifier == 'milestones':
-                    raise RuntimeError(
-                        "ERROR: This milestone does not have a type set and this is required. "
+                    raise ValueError(
```

Comment:
I am not sure I am following the logic, but with `if rc.type in MILESTONE_TYPES:` as the if statement, doesn't that  mean that this block has a VALID type and not an invalid type?

Reply:
there is an `else:` statement before the `raise ValueError` on line 161.
the code block looks like:

```
            if rc.type:
                if rc.type in MILESTONE_TYPES:
                    mil.update({'type': rc.type})
                else:
                    raise ValueError(
                        "ERROR: The type you have specified is not recognized. "
                        "Please rerun your command adding '--type' "
                        f"and giving a type from this list: {MILESTONE_TYPES}")
```

Reply:
:+1: Thanks.  It was hard to see that, so better to check.